insonmnia/hub/server.go

```diff
@@ -650,6 +628,29 @@ func (h *Hub) Devices(ctx context.Context, request *pb.Empty) (*pb.DevicesReply,
 	return reply, nil
 }
 
+func (h *Hub) MinerDevices(ctx context.Context, request *pb.ID) (*pb.DevicesReply, error) {
+	h.mu.Lock()
+	defer h.mu.Unlock()
```

Comment:
Looks like this lock is held a bit longer than needed.
Looks like h.getMinerByID fits here

Reply:
Fixed.